lib/ansible/modules/windows/setup.ps1

```diff
@@ -318,11 +323,12 @@ if($gather_subset.Contains('platform')) {
 
     $ansible_facts += @{
         ansible_architecture = $win32_os.OSArchitecture
-        ansible_domain = $ip_props.DomainName
-        ansible_fqdn = ($ip_props.Hostname + "." + $ip_props.DomainName)
-        ansible_hostname = $env:COMPUTERNAME
+        ansible_domain = $domainSuffix
+        ansible_fqdn = ($win32_cs.DNSHostname + "." + $domainSuffix)
```

Comment:
This will still have the same issue where if there is no domain suffix then the value is `"hostname."`. We need to fix it up so it only return the hostname if there is no domain. Also because `$domainSuffix` has the leading `.` we would now have `hostname..domain.com`.